### DISCUSSION TITLE
Queries about SCPH calculation

### DISCUSSION THREAD

**Discussion by ballal09** [ttadano/alamode#161] [General]
Dear Dr. Tadano,
I hope you are doing well. I have some queries about the SCPH calculation and evaluation of anharmonic force constants.
I have read the relevant   Q&A in this forum regarding the SCPH calculation and found the discussion on the thread (scph tutorial #80)  very useful. Steps for obtaining the anharmonic IFCs from the above mentioned thread are as follows:

1. Perform aimd and select a number of structures that are equally spaced in time.
2. Displace each of those structures by 0.1 Angstrom and calculate all configurations by DFT
3. Extract/collect all xmls (I'm using vasp) from all DFT calculations into a DFSET file
4. Then estimate the force constants using enet or adaptive lasso.

First of all, from the relevant discussion threads of  SCPH , I have learned that we need the 2nd and 4th order IFCs  for SCPH calculation. 
For the 2nd order IFCs, we calculate it using the 0 K prototype structure (not AIMD snapshot) by generating random displacement (ALM: mode (suggest)) and fitting the corresponding force & displacement. 

For the 4th order anharmonic  IFCs, one  way is to perform AIMD simulation and select a number of structures that are equally spaced in time as stated in the step 1. For the step 2 , I have some confusion and queries as follows:
 
Q1. How do we apply the random displacement to the selected AIMD snapshot? Is it done by the ALM mode suggest ,like the way the displacement pattern for 2nd order IFCS are generated? If so, it generates a large number of disp. pattern for each of those AIMD snapshots as those snapshots are disordered (lack symmetry). 
Q2. Is there any way to calculate the 4th order anharmonic IFCs from the sufficient AIMD snapshots without applying the random displacement? Like preparing the force displacement data sets (DFSET_anharmonic) from those snapshots.

My next question is about the calculating the anharmonic phonon lifetime:
Q3. How do we calculate the 3rd order IFCs for the  phonon lifetime? is it from AIMD snapshots or  generating the displacement from 0 K prototype structure?

Thank you very much for your time.
Best
Ballal

**Comment by ttadano** [comment]
> Q1. How do we apply the random displacement to the selected AIMD snapshot? Is it done by the ALM mode suggest ,like the way the displacement pattern for 2nd order IFCS are generated? If so, it generates a large number of disp. pattern for each of those AIMD snapshots as those snapshots are disordered (lack symmetry).

You can do this by issuing the command below:
`python ${ALAMODE_ROOT}/tools/displace.py --VASP SPOSCAR0 -md vasprun_md.xml -e 1001:5000:40 --random --mag 0.05 --prefix random`

`SPOSCAR0` is the supercell structure file without displacement, and `vasprun_md.xml` is the VASP output from your AIMD simulation. Please run `python ${ALAMODE_ROOT}/tools/displace.py -h` for more details.

> Q2. Is there any way to calculate the 4th order anharmonic IFCs from the sufficient AIMD snapshots without applying the random displacement? Like preparing the force displacement data sets (DFSET_anharmonic) from those snapshots.

Yes, it is possible. If your AIMD simulation is done with an accurate DFT parameter set, you can directly use the AIMD trajectory to extract anharmonic force constants. 
In this case, please prepare `DFSET` as
`python ${ALAMODE_ROOT}/tools/extract.py --VASP SPOSCAR0 vasprun_md.xml > DFSET_AIMD`
and use `DFSET_AIMD` in ALM with `LMODEL = ols`. 
You may need a large number of snapshots to reach a convergence of force constants.
Also, the force constants are biased to fit the AIMD trajectory done at the given temperature.

If you want to sample snapshots from an AIMD trajectory without adding random displacements, please issue 
`python ${ALAMODE_ROOT}/tools/displace.py --VASP SPOSCAR0 -md vasprun_md.xml -e 1001:5000:40 --prefix snapshot`

> Q3. How do we calculate the 3rd order IFCs for the phonon lifetime? is it from AIMD snapshots or generating the displacement from 0 K prototype structure?

Both are possible. You can either use `PREFIX.pattern_ANHARM3` to generate systematically displaced configurations or use the AIMD sampling higher-energy configurations.

**Comment by ballal09** [comment]
Dear Dr. Tadano

Thank you so much for your quick response. It was very helpful. I have couple of additional queries regarding SCPH and anharmonic lifetime calculation. 

Q1. Does increasing the number of AIMD snapshots improves the fitting error for the 4th order FCs? FYI,  I am applying the random displacement  to the snapshots.
Q2.  How do we calculate the phonon lifetime taking into account the anharmonicity from the SCPH approach? We need 3rd order FCs for that. Let's assume I have obtained 3rd order FCs by generating the displacement from 0 K prototype structure like you did in you STO (PRB) paper. Then only way the anharmonic effect  come into play in the phonon lifetime by replacing the eigenvector and frequencies by the renormalized one from the SCPH approach. When I looked into your kappa.in file for the thermal conductivity of STO, I have noticed you used:

FCSXML = STO_anharm.xml 
FC2XMl = STO222_scph_300K.xml

Here, STO_anharm.xml  is the 4th order FCs obtained from the AIMD snapshot?  I guess , STO222_scph_300K.xml is the renormalized 2nd order FCs from the SCPH. 

Would you please explain how to obtain STO222_scph_300K.xml from the SCPH calculation?

Where is the 3rd order FCs used here?

Q3. Disregarding SCPH approach, how to study the effect of 4th order anharmonicity in the phonon lifetime using Alamode?

Thank you so much for your time.

Best
Ballal
